sbi/inference/abc/mcabc.py

```diff
@@ -158,13 +166,24 @@ def __call__(
         else:
             theta_adjusted = theta_accepted
 
-        posterior = Empirical(theta_adjusted, log_weights=ones(theta_accepted.shape[0]))
+        if kde:
+            self.logger.info(
+                f"""KDE on {theta_adjusted.shape[0]} samples with bandwidth option
+                {kde_bandwidth}. Beware that KDE can give unreliable results when used
+                with too few samples and in high dimensions."""
+            )
+
+            kde_dist = get_kde(
+                theta_adjusted, bandwidth=kde_bandwidth, transform=kde_transform
+            )
 
-        if return_distances and return_x_accepted:
-            return posterior, distances_accepted, x_accepted
-        if return_distances:
-            return posterior, distances_accepted
-        if return_x_accepted:
-            return posterior, x_accepted
+            if return_summary:
+                return kde_dist, dict(
+                    theta=theta_adjusted, distances=distances_accepted, x=x_accepted
+                )
+            else:
+                return kde_dist
+        elif return_summary:
+            return theta_accepted, dict(distances=distances_accepted, x=x_accepted)
```

Comment:
I think we'd want to return potentially adjusted thetas here rather than `theta_accepted`?